src/NUnitFramework/framework/Constraints/ThrowsNothingConstraint.cs

```diff
@@ -50,23 +50,9 @@ public override string Description
         /// <returns>True if no exception is thrown, otherwise false</returns>
         public override ConstraintResult ApplyTo<TActual>(TActual actual)
         {
-            //TestDelegate code = actual as TestDelegate;
-            //if (code == null)
-            //    throw new ArgumentException("The actual value must be a TestDelegate", "actual");
-
-            //caughtException = null;
-
-            //try
-            //{
-            //    code();
-            //}
-            //catch (Exception ex)
-            //{
-            //    caughtException = ex;
-            //}
-
-            caughtException = ThrowsConstraint.ExceptionInterceptor.Intercept(actual);
-
+            caughtException = actual as Exception;
+            if (caughtException == null && actual != null)
+                caughtException = ThrowsConstraint.ExceptionInterceptor.Intercept(actual);
```

Comment:
The most obvious refactoring I can think of is to create a new `Constraint` class that would check that the exception is null, and fail with the same message as `ThrowsNothingConstraint` otherwise, but I couldn't think of a good name; what bothers me is that the message wouldn't match what the constraint is actually checking.